File: BubbleSort.cpp
/*@author Priti Chattopadhyay
  @date: 17th March, 2019
  @day_time Sunday 14:30
*/

using namespace std;
//Preprocessor directive
#include<iostream>

//Bubble Sort in C++
void bubbleSort(int*, int);
void display(int*, int);

int main()
{
	// Take an array of n integers 
	int a[5] = {8, 6, 4, 9, 2};
	cout<<"\nArray before Sorting:\n";
	//calling the routine to display elements in the array before sorting
	display(a, 5);
	//calling a routine to sort the elements
	bubbleSort(a, 5);
	return 0;
}

//Routine to sort the elements of the array
//Sorting in non - decreasing order
void bubbleSort(int *arr, int n)
{
	int i, j, temp;
	//Outer loop
	for(i = 0; i < n-1; i++)
	{
		//Inner loop
		for(j = 0; j < n-1-i; j++)
		{
			//Compare two adjacent elements
			if(arr[j] > arr[j + 1])
			{
				//swap a[j] and a[j+1] when true
				temp = arr[j];
				arr[j] = arr[j + 1];
				arr[j + 1] = temp;		
			}//end if
		}//end Inner loop
	}//end Outer loop
	
	cout<<"\nArray after sorting:\n";
	//calling a routine to display the elements of the array after sort
	display(arr, n);
}

// Routine to display elements of an array 
void display(int *a, int n)
{
	int i;
	for(i = 0; i < n; i++)
	{
		cout<<a[i]<<"\t";
	}
	cout<<endl<<endl;
}

File: BubbleSort.java
public class BubbleSort{
/*@author Priti Chattopadhyay
  @date: 31th March, 2019
  @day_time Sunday 11:50
*/
	//Bubble sort in Java
	// Declaration of class BubbleSort
	public static void main(String args[]){
		//Initializing an array of 5 elements
		int a[] = {8, 6, 4, 9, 2};
		//To get the length of the array
		int n = a.length;
		//Creating an object of class BubbleSort to call routines or methods
		BubbleSort obj = new BubbleSort();
		System.out.println("\nArray before sorting:");
		//calling a routine to display elements of the array before sorting
		obj.displayArray(a, n);//In Java, we always pass a reference to an array
		//calling the routine to sort the elements using bubble sort algorithm
		obj.bubbleSort(a, n);
		System.out.println("\nArray after sorting");
		//calling a routine to display elements of the array after sorting
		obj.displayArray(a, n);
	}
	
	//Routine to sort the array using bubble sort algorithm
	//Sorting the elements in non-decreasing order
	void bubbleSort(int a[], int n){
		int temp;//a temporary variable used during swapping elements
		//Outer loop
		for(int i = 0; i < n - 1; i++){
			//Inner loop
			for(int j = 0; j < n - 1 - i; j++){
				//Comparing adjacent elements
				if(a[j] > a[j + 1]){
					//Swapping of elements present at a[j] and a[j+1]
					temp = a[j];
					a[j] = a[j + 1];
					a[j + 1] = temp;
				}//if ends
			}//Inner loop ends
		}//Outer loop ends
	}//Bubble sort routine ends

	//Routine to display elements of an array
	void displayArray(int a[], int n){
		for(int i = 0; i < n; i++){
			System.out.print(a[i] + "\t");
		}
		System.out.println("\n");
	}//end of display routine
}
